models/address_book.rb

```diff
@@ -18,4 +19,18 @@ def add_entry(name, phone, email)
 
     @entries.insert(index, Entry.new(name, phone, email))
   end
+
+  def remove_entry(name, phone, email)
+    # using index where I could have set a delete_entry variable
+    index = 0
+
+    @entries.each do |entry|
+      if name == entry.name && phone = entry.phone_number && email == entry.email
```

Comment:
You might be getting some false positives here :smiley:. The second condition is missing an extra `=`:

```ruby
phone == entry.phone_number
```

Currently, it will return true even if the phone doesn't match. The interpreter treats it as an assignment, so `phone` will be "truthy" - yeah that's a term in Ruby :stuck_out_tongue: - as long as `entry.phone_number` returns a non-falsy value (that's also a term).